language/move-stdlib/src/doc/mod.rs

```diff
@@ -0,0 +1,166 @@
+    mapping
+        .iter()
+        .map(|(name, addr)| (name.to_string(), NumericalAddress::parse_str(addr).unwrap()))
+        .collect()
+}
+
+pub fn build_doc(
+    output_path: &str,
+    doc_path: &str,
+    templates: Vec<String>,
+    references_file: Option<String>,
+    sources: &[String],
+    dep_paths: Vec<String>,
+    with_diagram: bool,
+    named_addresses: BTreeMap<String, NumericalAddress>,
+) {
+    let options = move_prover::cli::Options {
+        move_sources: sources.to_vec(),
+        move_deps: dep_paths,
+        move_named_address_values: move_prover::cli::named_addresses_for_options(&named_addresses),
+        verbosity_level: LevelFilter::Warn,
+        run_docgen: true,
+        docgen: move_docgen::DocgenOptions {
+            root_doc_templates: templates,
+            references_file,
+            doc_path: vec![doc_path.to_string()],
+            output_directory: output_path.to_string(),
+            include_dep_diagrams: with_diagram,
+            include_call_diagrams: with_diagram,
+            ..Default::default()
+        },
+        ..Default::default()
+    };
+    options.setup_logging_for_test();
+    move_prover::run_move_prover_errors_to_stderr(options).unwrap();
+}
+
+pub fn build_stdlib_doc(output_path: &str) {
+    build_doc(
+        output_path,
+        "",
+        vec![path_in_crate(OVERVIEW_TEMPLATE)
+            .to_string_lossy()
+            .to_string()],
+        Some(
+            path_in_crate(REFERENCES_TEMPLATE)
+                .to_string_lossy()
+                .to_string(),
+        ),
+        move_stdlib_files().as_slice(),
+        vec![],
+        false,
+        move_stdlib_named_addresses(),
+    )
+}
+
+pub fn build_nursery_doc(output_path: &str) {
+    build_doc(
+        output_path,
+        "",
+        vec![],
+        None,
+        move_nursery_files().as_slice(),
+        vec![move_stdlib_modules_full_path()],
+        false,
+        move_stdlib_named_addresses(),
+    )
+}
+
+pub fn build_error_code_map(output_path: &str) {
+    let options = move_prover::cli::Options {
+        move_sources: move_stdlib_files(),
+        move_deps: vec![],
+        verbosity_level: LevelFilter::Warn,
+        run_errmapgen: true,
+        move_named_address_values: move_prover::cli::named_addresses_for_options(
+            &move_stdlib_named_addresses(),
+        ),
+        errmapgen: move_errmapgen::ErrmapOptions {
+            output_file: output_path.to_string(),
+            ..Default::default()
+        },
+        ..Default::default()
+    };
+    options.setup_logging_for_test();
+    move_prover::run_move_prover_errors_to_stderr(options).unwrap();
+}
+
+const ERROR_DESCRIPTIONS: &[u8] = if cfg!(feature = "address20") {
+    include_bytes!("../../address20-error_description.errmap")
+} else if cfg!(feature = "address32") {
+    include_bytes!("../../address32-error_description.errmap")
+} else {
+    include_bytes!("../../error_description.errmap")
```

Comment:
@asmie some review tips:

The updated paths here are (adding an extra `../`), I think, the only changes in this file.
It was mostly copy/paste from `lib.rs` to `mod.rs`

Reply:
Thanks! 😄